File: IntegerToRoman/itr.py
class Solution:
    def intToRoman(self, num: int) -> str:
        dic = {1000:'M',
               900:'CM',
               500:'D',
               400:'CD',
               100:'C',
               90:'XC',
               50:'L',
               40:'XL',
               10:'X',
               9:'IX',
               5:'V',
               4:'IV',
               1:'I'}
        res = ""
        for key in dic.keys():
            res += dic[key] * (num//key)
            num %= key
        return res

File: IntegerToRoman/itr.java
class Solution {
    public String intToRoman(int num) {
        int[] keys = {1000,900,500,400,100,90,50,40,10,9,5,4,1};
        String[] values = {"M","CM","D","CD","C","XC","L","XL","X","IX","V","IV","I"};
        StringBuilder res = new StringBuilder();
        for(int i=0;i<keys.length;i++){
            while(num >= keys[i]){
                res.append(values[i]);
                num -= keys[i];
            }
        }
        return res.toString();
    }
}